commit 2fadaef41283aad7100fa73f01998cddaca25833
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Tue Feb 15 22:52:11 2011 -0500

    auxv: require the target to be tracable (or yourself)
    
    same as for environ, except that we didn't do any checks to
    prevent access after suid execve
    
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>

diff --git a/fs/proc/base.c b/fs/proc/base.c
index fc471b8..e94b58b 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -281,9 +281,9 @@ out:
 
 static int proc_pid_auxv(struct task_struct *task, char *buffer)
 {
-	int res = 0;
-	struct mm_struct *mm = get_task_mm(task);
-	if (mm) {
+	struct mm_struct *mm = mm_for_maps(task);
+	int res = PTR_ERR(mm);
+	if (mm && !IS_ERR(mm)) {
 		unsigned int nwords = 0;
 		do {
 			nwords += 2;
